### PR TITLE
[canvaskit] Add animation detection for GIFs

### PR DESCRIPTION
**Pull request by harryterkelsen**
Detect if a GIF is animated to determine if we need to use Skia to decode it or if we can use  tag decoding.

Fixes https://github.com/flutter/flutter/issues/151911

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by kevmoo** [comment]
Nit: I wish the source pointed to some docs about how this all was interpreted.

Would ease maintenance later.

**Comment by harryterkelsen** [comment]
I linked to the GIF 89a spec in the doc comment for the class. Unfortunately since it's a text file I can't link to specific sections corresponding to the methods, but I stick to the terminology in the spec so the sections should be easy to find if you have the spec open.

File: lib/web_ui/lib/src/engine/image_format_detector.dart [hunk 1]
```diff
@@ -217,28 +227,22 @@ class _WebpHeaderReader {
...
     return chunkFourCC == expectedHeader;
   }
 
   /// Reads the WebP header. Returns [false] if this is not a valid WebP header.
   bool _readWebpHeader() {
     final String riffBytes = _readFourCC();
 
-    // Read file size byte.
-    _readUint32();
+    // Skip reading file size byte.
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/image_format_detector.dart:239]: nit:
```suggestion
    // Skip reading file size bytes.
```

**Review by mdebbar — APPROVED** [review]
LGTM!

File: lib/web_ui/lib/src/engine/image_format_detector.dart (hunk 1 above)
Comment (harryterkelsen) [reply lib/web_ui/lib/src/engine/image_format_detector.dart:239]: Done.

**Pull request merged**